src/openforms/submissions/models/submission_files.py

```diff
@@ -190,15 +190,65 @@ def informatieobjecttype(self) -> Optional[str]:
         """
         Get the informatieobjecttype for this attachment from the configuration
         """
-        formdef_config = self.submission_step.form_step.iter_components()
-        for component in formdef_config:
+        # use configuration wrapper for caching
+        formdef_config = (
+            self.submission_step.form_step.form_definition.configuration_wrapper.configuration
+        )
+        for component in formdef_config["components"]:
```

Comment:
this is wrong since the `configuration` is basically an alias of the underlying formio configuration.

Instead you should loop over the `FormioConfigurationWrapper` instance, which properly recurses while using caches.